swift/src/CompileSlicePlugin/CompileSlicePlugin.swift

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) ZeroC, Inc. All rights reserved.
+
+import Foundation
+import PackagePlugin
+
+enum PluginError: Error {
+    case invalidTarget(Target)
+    case missingCompiler(String)
+    case missingConfigFile(String, String)
+    case missingIceSliceFiles(String)
+
+    var description: String {
+        switch self {
+        case .invalidTarget(let target):
+            return "Expected a SwiftSourceModuleTarget but got '\(type(of: target))'."
+        case .missingCompiler(let path):
+            return "Missing slice compiler: '\(path)'."
+        case .missingConfigFile(let path, let target):
+            return
+                "Missing config file '\(path)` for target `\(target)`. '. This file must be included in your sources."
+        case .missingIceSliceFiles(let path):
+            return "The Ice slice files are missing. Expected location: `\(path)`"
+        }
+    }
+}
+
+/// Represents the contents of a `slice-plugin.json` file
```

Comment:
We need a better documentation for this CompileSlicePlugin here, and also in a README file.

Reply:
Yes.